radicle-httpd/src/api/v1/projects.rs

```diff
@@ -166,14 +161,23 @@ async fn history_handler(
         })
         .skip(page * per_page)
         .take(per_page)
-        .filter_map(|commit| {
-            if let Ok(commit) = commit {
-                source::commit(&repo, commit.id).ok()
-            } else {
-                None
-            }
+        .map(|r| {
+            r.and_then(|c| {
+                let glob = Glob::all_heads().branches().and(Glob::all_remotes());
+                let branches: Vec<String> = repo
+                    .revision_branches(c.id, glob)?
+                    .iter()
+                    .map(|b| b.refname().to_string())
+                    .collect();
+                let diff = repo.diff_commit(c.id)?;
+                Ok(json!({
+                    "header": api::json::commit(&c),
+                    "diff": diff,
+                    "branches": branches
+                }))
```

Comment:
Not a blocker, but maybe all `json!` calls should go into the `json` module now

Reply:
aye, this can be cleaned up later.